crates/bevy_math/src/cubic_splines.rs

```diff
@@ -1055,12 +1062,31 @@ impl CubicSegment<Vec2> {
     }
 }
 
-/// A collection of [`CubicSegment`]s chained into a single parametric curve. Has domain `[0, N)`
-/// where `N` is the number of attached segments.
+impl<P: VectorSpace> Curve<P> for CubicSegment<P> {
+    #[inline]
+    fn domain(&self) -> Interval {
+        Interval::UNIT
+    }
+
+    #[inline]
+    fn sample_unchecked(&self, t: f32) -> P {
+        self.sample_clamped(t)
```

Comment:
Why the call to sample_clamped, we can just do:
```suggestion
        self.position(t)
```

Reply:
I'm too used to keyframe curves. You're right — we should just do that everywhere and allow extrapolation, probably.